lib/block.mli

```diff
@@ -34,6 +34,21 @@ val blkgetsize: string -> Unix.file_descr -> [ `Ok of int64 | `Error of error ]
 
 val connect : string -> [`Ok of t | `Error of error] io
 
+type config = {
+  buffered: bool; (** true if I/O hits the OS disk caches, false if "direct" *)
+  sync: bool; (** true if [flush] flushes all caches, including disk drive caches *)
+  path: string; (** path to the underlying file *)
+}
+(** Configuration of a device *)
+
+val connect_uri : Uri.t -> [`Ok of t | `Error of error] io
```

Comment:
I think there should be a separate function (e.g. `parse_config : string -> config or_error`) either here or in the code that uses this. We shouldn't force other users of the API to turn bools into URIs with query strings.